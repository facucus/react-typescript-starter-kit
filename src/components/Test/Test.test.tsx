import React from "react";
import { render, fireEvent } from "react-testing-library";
import Test from "./Test";

describe("Testing Test component", () => {
  const defaultProps = {
    text: "This is my test",
    onClick: jest.fn()
  };
  test("Checking correct Title", () => {
    const { getByTestId } = render(<Test {...defaultProps} />);

    expect(getByTestId("title-text")).toHaveTextContent("Hola Mundo");
  });

  test("Checking correct Title", () => {
    const { container } = render(<Test {...defaultProps} />);

    const text = container.querySelector("p");
    expect(text).toHaveTextContent("This is my test");
  });

  test("Button click", () => {
    const { getByText, debug } = render(<Test {...defaultProps} />);
    const button = getByText(/click me/i);
    fireEvent.click(button);
    expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
  });

  test("Match Snapshot", () => {
    const { container } = render(<Test {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
