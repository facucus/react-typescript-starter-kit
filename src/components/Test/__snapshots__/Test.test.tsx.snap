// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Test component Match Snapshot 1`] = `
<div>
  <h1
    data-testid="title-text"
  >
    Hola Mundo
  </h1>
  <p
    class="text"
  >
    This is my test
  </p>
  <button>
    Click Me
  </button>
</div>
`;
